### PR TITLE
[UI] Pass namespace to APIs

### PR DESCRIPTION
**Pull request by Bobgy**
Part of multi user support.

Changes:
* Pass context got from kubeflow central dashboard to KFP UI
* When creating a run, namespace is added as a resource reference
* When getting log from pods, namespace is passed to find the pod by (name, namespace)
* Refactor unit tests because new context api doesn't work with enzyme shallow testing, changed to mounting complete instances https://github.com/kubeflow/pipelines/issues/2732.

TODO:
* [x] add unit tests

/cc @gaoning777 
/area frontend
/kind feature

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2676)

### CODE REVIEW

**Comment by Bobgy** [comment]
@gaoning777 This is ready for review.

File: frontend/server/k8s-helper.ts [hunk 1]
```diff
@@ -184,11 +184,11 @@ export function waitForTensorboardInstance(logdir: string, timeout: number): Pro
   });
 }
 
-export function getPodLogs(podName: string): Promise<string> {
+export function getPodLogs(podName: string, podNamespace?: string): Promise<string> {
   if (!k8sV1Client) {
     throw new Error('Cannot access kubernetes API');
   }
-  return (k8sV1Client.readNamespacedPodLog(podName, namespace, 'main') as any).then(
+  return (k8sV1Client.readNamespacedPodLog(podName, podNamespace || namespace, 'main') as any).then(
```
Comment (gaoning777) [inline frontend/server/k8s-helper.ts]: Do we need to update the global variable: namespace such that the getArgoWorkflow and getK8sSecret also work?

Comment (Bobgy) [reply frontend/server/k8s-helper.ts]: We need to support those one by one. Global variable is shared and could be modified during processing of a request.

I don't currently know where those methods are used. I guess I should double check and make sure other features don't break.

Comment (gaoning777) [reply frontend/server/k8s-helper.ts]: e.g. I searched that https://github.com/kubeflow/pipelines/blob/master/frontend/server/workflow-helper.ts#L147 is the usage of the workflow-helper calling the getArgoWorkflow. Could you proof-check if this functions are called in other places.

Comment (Bobgy) [reply frontend/server/k8s-helper.ts]: Thanks, that was added in https://github.com/kubeflow/pipelines/pull/2081 to get pod log using argo workflow log persistence capability. I don't think we need to support this in multi user mode right now

Comment (Bobgy) [reply frontend/server/k8s-helper.ts]: I verified these methods are only used for the use case I mentioned above. It's fine we defer these support.

Comment (gaoning777) [reply frontend/server/k8s-helper.ts]: SG

**Pull request merged**